app/models/library_entry.rb

```diff
@@ -200,12 +200,28 @@ def sync_to_mal?
 
     media.trending_vote(user, 0.5) if progress_changed?
     media.trending_vote(user, 1.0) if status_changed?
+  end
+
+  # TODO: will rename this if I think of a better one
+  after_commit :create_or_update_library_entry, on: %i[create update]
 
-    # Sync MAL updates if linked profile exists
-    MyAnimeListSyncWorker.perform_async(id, 'create/update') if sync_to_mal?
+  def create_or_update_library_entry
+    if sync_to_mal?
```

Comment:
You just told me like 2 lines above that I should not use a `return if` when it is a multi-line statement...